protocols/migration/participants.py

```diff
@@ -583,3 +584,158 @@ def migrate_tumour_sample(self, tumour_sample):
         return self.validate_object(
             object_to_validate=migrated_tumour_sample, object_type=self.new_model.TumourSample
         )
+
+
+class MigrationParticipants100ToReports(BaseMigration):
+    old_model = participant_1_0_0
+    new_model = participant_old
+
+    def migrate_pedigree(self, old_pedigree):
+        """
+        :param old_pedigree: org.gel.models.participant.avro.Pedigree 1.0.0
+        :return: org.gel.models.report.avro RDParticipant.Pedigree 3.0.0
+        """
+        new_pedigree = self.convert_class(self.new_model.Pedigree, old_pedigree)
+        new_pedigree.versionControl = self.new_model.VersionControl()
+        new_pedigree.gelFamilyId = old_pedigree.familyId
+        new_pedigree.analysisPanels = self.migrate_analysis_panels(old_panels=old_pedigree.analysisPanels)
+        new_pedigree.participants = self.migrate_members_to_participants(old_members=old_pedigree.members,
+                                                                         family_id=old_pedigree.familyId)
+        return self.validate_object(object_to_validate=new_pedigree, object_type=self.new_model.Pedigree)
+
+    def migrate_members_to_participants(self, old_members, family_id):
+        return [self.migrate_member_to_participant(old_member=old_member, family_id=family_id) for old_member in
+                old_members]
+
+    def migrate_member_to_participant(self, old_member, family_id):
+        new_participant = self.convert_class(self.new_model.RDParticipant, old_member)
+        new_participant.gelFamilyId = family_id
+        new_participant.pedigreeId = old_member.pedigreeId
+        new_participant.isProband = old_member.isProband or False
+        new_participant.sex = self.migrate_sex(old_sex=old_member.sex)
+        new_participant.personKaryotipicSex = self.migrate_person_karyotypic_sex(old_pks=old_member.personKaryotypicSex)
+        new_participant.yearOfBirth = str(old_member.yearOfBirth)
+        new_participant.adoptedStatus = self.migrate_adopted_status(old_status=old_member.adoptedStatus)
+        new_participant.lifeStatus = self.migrate_life_status(old_status=old_member.lifeStatus)
+        new_participant.affectionStatus = self.migrate_affection_status(old_status=old_member.affectionStatus)
+        new_participant.hpoTermList = self.migrate_hpo_term_list(old_list=old_member.hpoTermList)
+        new_participant.samples = self.migrate_samples(old_samples=old_member.samples)
+        new_participant.versionControl = self.new_model.VersionControl()
+
```

Comment:
if old_member.consentStatus == None then the model will be invalid as it is not nullable in reports 3.0.0. I would suggest to add:
```
if old_model.consentStatus is None:
    new_model.consentStatus = self.new_model.ConsentStatus(programmeConsent=True, primaryFindingConsent=True, secondaryFindingConsent=True, carrierStatusConsent=True)
```